File: go/wasm/bech32.go
// +build js,wasm

package main

import (
	"github.com/cosmos/amino-js/go/src"
	"syscall/js"
)

func EncodeBech32(this js.Value, args []js.Value) interface{} {
	bz := typedArrayToByteSlice(args[1])
	bech, err := src.EncodeBech32(args[0].String(), bz)
	return []interface{}{bech, errorOf(err)}
}

func DecodeBech32(this js.Value, args []js.Value) interface{} {
	hrp, data, err := src.DecodeBech32(args[0].String())
	return []interface{}{hrp, js.TypedArrayOf(data), errorOf(err)}
}

File: src/wasm/bech32.ts
import * as Amino from '../../lib/Amino.wasm';
import { Bech32String, Bytes } from '../../lib/types';

export function encodeBech32 (hrp: string, data: Bytes): Bech32String {
    const [bech, error] = Amino.encodeBech32(hrp, data);
    if (error) {
        throw new Error(error);
    }
    return bech;
}

export function decodeBech32 (bech: Bech32String): [string, Bytes] {
    const [hrp, data, error] = Amino.decodeBech32(bech);
    if (error) {
        throw new Error(error);
    }
    return [hrp, data];
}
